### PR TITLE
Move all subscriptions services to the OrderManagement engine

### PR DESCRIPTION
**Pull request by luisramos0**
#### What? Why?

The StripeSCA project will involve some work related to payments on subscriptions.
I started by refactoring the subs confirmation job where I'll have to add a few things related to 
payments but ended up moving all the subscriptions services to the OM engine.

I have moved 3 services from lib/open_food_network :tada: and 5 from app/services.
The services are all now under engines/order_management/app/services/order_management/subscriptions

I am happy to get a little step more into the "break OFN into domains" project.

There are no functional changes whatsoever in the PR. It's 100% refactoring.

Apart from refactoring the subs_confirmation_job and moving the files to the OM engine, a few rubocop issues were fixed.

#### What should we test?
We need to validate that subscriptions are still working.

#### Release notes
Changelog Category: Changed
Refactored and moved subscriptions code to improve organization and make future work on subscriptions easier.

### CODE REVIEW

File: engines/order_management/app/services/order_management/subscriptions/count.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module OrderManagement
+  module Subscriptions
+    class Count
```
Comment (sauloperez) [inline engines/order_management/app/services/order_management/subscriptions/count.rb]: :ok_hand: :heart:

**Comment by sauloperez** [comment]
ping @luisramos0 :point_up:

**Comment by luisramos0** [comment]
I rebased to resolve conflicts. I had to manually replay some code on the moved files :-(
Anyway, done.

**Comment by luisramos0** [comment]
I didnt try the tests before rebasing.... after rebasing the subscription test is not failing, only the enterprise_fee_summary report specs.

That means this PR should be ok to go forward.
I created #4958 for the specs. We need to fix this now.

**Comment by sauloperez** [comment]
thoughts @mkllnk @Matt-Yorkley ?

**Comment by luisramos0** [comment]
I just double checked and confirmed the order_management specs are already broken in master.

**Comment by Matt-Yorkley** [comment]
@luisramos0 I think we need to be absolutely sure that all of our engine specs are running correctly before moving all of subscriptions into an engine...

I've unassigned myself from #4958 if you want to pick it up. I couldn't get the engine specs to play nicely with knapsack.

**Comment by luisramos0** [comment]
I rebased this again to resolve conflicts.

**Comment by luisramos0** [comment]
@Matt-Yorkley the engine specs issue is solved now.
Can you have a look at this please? Thanks!

File: engines/order_management/spec/performance/order_management/subscriptions/proxy_order_syncer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module OrderManagement
+  module Subscriptions
+    describe ProxyOrderSyncer, performance: true do
```
Comment (Matt-Yorkley) [inline engines/order_management/spec/performance/order_management/subscriptions/proxy_order_syncer_spec.rb]: Spec with the :performance tag don't actually get run in CI, right?

I guess these are just being moved across though... :+1:

Comment (luisramos0) [reply engines/order_management/spec/performance/order_management/subscriptions/proxy_order_syncer_spec.rb]: yes, just moving. I thought we could delete these, but they are harmless and can be useful in the future. I was not sure so I left them...

**Comment by filipefurtad0** [comment]
Hi @luisramos0,

Tested the subscription functionality following the test-plan from the testing handbook, all seems to work as expected. Specifically,

- created a subscription
- verified that orders are placed
- confirmation emails are sent
- order shows up in "Transactions" (customer)

I think this is good to go.

**Comment by luisramos0** [comment]
hurray! thanks Filipe!

**Pull request merged**